src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1370,6 +1371,7 @@ private void RerouteAllQueuedMessages(ActivationData activation, ActivationAddre
                     {
                         RequestContext.Clear(); // Clear any previous RC, so it does not leak into this call by mistake. 
                         await activation.GrainInstance.OnDeactivateAsync();
+                        await activation.LifeCycle.OnStop();
```

Comment:
Perhaps it's acceptable, but if the grain throws in OnDeactivateAsync, then OnStop will never be called.